homeassistant/components/tesla/__init__.py

```diff
@@ -310,4 +310,5 @@ def refresh(self) -> None:
         This assumes the coordinator has updated the controller.
         """
         self.tesla_device.refresh()
+        self._attributes = self.tesla_device.attrs.copy()
```

Comment:
We're copying this attributes here and then again the the `device_state_attributes` property. Probably remove the latter copy.

Side note: We want to set the state attribute keys specifically in `device_state_attributes` and not forward the data directly from the library. We want to use a dict that maps between the home assistant keys and the library data keys, and iterate over this dict when setting the attribute items, to not couple the two different APIs directly. We can fix this in a follow up PR.

Reply:
Latter copy is removed.